arangod/VocBase/AuthInfo.cpp

```diff
@@ -757,28 +767,57 @@ AuthLevel AuthInfo::canUseCollection(std::string const& username,
   if (coll[0] >= '0' && coll[0] <= '9') {
     // lookup by collection id
     // translate numeric collection id into collection name
-    return canUseCollectionInternal(username, dbname, DatabaseFeature::DATABASE->translateCollectionName(dbname, coll));
+    std::string tmpCol = DatabaseFeature::DATABASE->translateCollectionName(dbname, coll);
+    if (tmpCol.empty()) {
+      // no collection name given
+      return AuthLevel::NONE;
+    }
+ 
+    loadFromDB();
+    READ_LOCKER(guard, _authInfoLock);
+    return canUseCollectionInternal(username, dbname, tmpCol);
+  }
+
+  loadFromDB();
+  READ_LOCKER(guard, _authInfoLock);
+  // lookup by collection name
+  return canUseCollectionInternal(username, dbname, coll);
```

Comment:
Why have this twice ?

Reply:
because of the temporary string that is created in the "numeric" case but is not in the "name" case.
We save a copy here, and as we often call this function it may be a optimization worth it.